doc/specs/#980 - Apps and Features entries version mapping.md

```diff
@@ -0,0 +1,193 @@
+---
+author: Yao Sun @yao-msft
+created on: 2022-05-09
+last updated: 2022-05-15
+issue id: 980
+---
+
+# Mapping of Apps and Features version and Winget package version
+
+For [#980](https://github.com/microsoft/winget-cli/issues/980)
+
+## Abstract
+
+Some Winget packages may have the concept of marketing version and internal version. So they may have different version values in Winget manifest and in Apps and Features registry entry. Usually, the marketing version will be in the Winget manifest and internal version will be written in the Apps and Features registry.  
+
+In this doc, Winget manifest version will be referred to as Winget version and Apps and Features version in registry will be referred to as ARP version.
+
+## Inspiration
+
+Winget should try to map or correlate the ARP version of an installed package to its Winget version. This will help Winget List/Upgrade better determine if there's an upgrade available.
+
+## Solution Design
+
+### Version parsing and comparison logic in Winget
+
+Versions are parsed by:
+1. Splitting the string based on the split character (`.`)
+2. Parsing a leading, positive integer from each split part
+3. Saving any remaining, non-digits as a supplemental value
+
+Versions are compared by:  
+ for each part in each version  
+  if both sides have no more parts, return equal  
+  else if one side has no more parts, it is less  
+  else if integers not equal, return comparison of integers  
+  else if only one side has a non-empty string part, it is less  
+  else if string parts not equal, return comparison of strings  
+
+For example:  
+Version `1` is less than version `2`  
+Version `1.0.0` is less than version `2.0.0`  
+Version `0.0.1-alpha` is less than version `0.0.2-alpha`  
+Version `0.0.1-beta` is less than version `0.0.2-alpha`  
+Version `0.0.1-alpha` is less than version `0.0.1-beta`  
+Version `0.0.1-alpha` is less than version `0.0.1` 
+Version `13.9.8` is less than version `14.0`  
+Version `1.0` is equal to version `1.0.0`  
```

Comment:
I don't know the answer, but based on "else if one side has no more parts, it is less ", `1.0` would be less than `1.0.0`.

Reply:
I agree here, I think that the no-more-parts check should consider `0` the same as an undefined part in terms of equality

Reply:
These examples are actually adapted from our unit tests, so these are what we are currently using for version comparison now. During parsing, version parts with 0 only get dropped. That's why 1.0.0 == 1.0. I'll update the doc to clarify that.